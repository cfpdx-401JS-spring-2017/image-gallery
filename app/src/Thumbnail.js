import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Thumbnail extends Component {

  static propTypes = {
    data: PropTypes.array.isRequired
  }

  render() {
    const { data } = this.props;

    return (
      <ul>
        {data.map((img, i) => (
          <li key={i} className='Thumbnail'>
            <div>
              <img src={img.url} alt={img.title} className='Thumbnail-image'/>
            </div>
            <div className='Thumbnail-title'>
              {img.title}
            </div>
          </li>
        ))}
      </ul>
    );
  }
}

export default Thumbnail;