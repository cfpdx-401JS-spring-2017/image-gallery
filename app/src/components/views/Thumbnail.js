import React from 'react';
import PropTypes from 'prop-types';
import ThumbnailImage from './ThumbnailImage';

Thumbnail.propTypes = {
  poms: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
    url: PropTypes.string.isRequired
  }))
};

export default function Thumbnail({ poms, handleDelete }) {
  return (
    <ul>
      {poms.map((pom, i) => <ThumbnailImage
        key={pom._id}
        {...pom}
        onDelete={() => handleDelete(pom._id, i)}
      />)}
    </ul>
  );
}