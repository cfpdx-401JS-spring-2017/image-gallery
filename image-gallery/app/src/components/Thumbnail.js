import React from 'react';
import PropTypes from 'prop-types';
import DeleteImage from './DeleteImage';



export default function Thumbnail(props) {


    return (
        <div>
            {props.images.map((image, index) => (
                <figure key={index}
                    style={{
                        display: 'inline-block'
                    }}>
                    <a href={image.wiki}> <img className='thumbnail-image' src={image.url} alt="skateboarders"
                        style={{
                            height: '100px',
                            width: '100px',
                            transition: 'all .2s ease-in-out'
                        }}
                    />
                    </a>
                    <figcaption className='image-caption'> {image.title} </figcaption>
                    <DeleteImage removeImage={() => props.removeImage(props.albumId, image._id)} />
                </figure>
            ))}
        </div>

    )

}

Thumbnail.proptypes = {
    images: PropTypes.arrayOf(PropTypes.object).isRequired

};  

