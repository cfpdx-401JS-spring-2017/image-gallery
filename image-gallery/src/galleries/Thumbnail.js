import React, { Component } from 'react';

export default class Thumbnail extends Component {

    render() {
        return (
            <div>
                <h2>THUMBNAIL VIEW</h2>
                {this.props.images.map(image => (
                    <figure style={{
                        display:'inline-block'}}>
                    <img src={image.url} style={{
                        height: '100px',
                        width: '100px',
                    }}
                    />
                     <figcaption> {image.title} </figcaption>
                     </figure>   
                ))}
            </div>

        )
    }
}

