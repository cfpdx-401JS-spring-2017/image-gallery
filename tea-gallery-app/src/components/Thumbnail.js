import React from 'react';
import PropTypes from 'prop-types';
import ThumbnailItem from './ThumbnailItem';

Thumbnail.PropTypes = {
  images: PropTypes.array.required,
  onDeleteImage: PropTypes.func.required
};

export default function Thumbnail(props) {

  return (
    <div>
      <ThumbnailItem 
      images={props.images}
      onDeleteImage={props.onDeleteImage}
      />
    </div>
  );
}

