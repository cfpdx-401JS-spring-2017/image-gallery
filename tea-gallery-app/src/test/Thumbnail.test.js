import React from 'react';
import renderer from 'react-test-renderer';
import Thumbnail from '../components/Thumbnail';

const images = [
  {
    title: 'tea leaves in teapot',
    description: 'what nice leaves',
    url: '/../public/img/teainpot.jpg'
  },
  {
    title: 'Tetsubin over charcoal',
    description: 'what a nice tetsubin',
    url: '/../public/img/tetsubin-charcoal.jpg'
  },
  {
    title: 'Teapot display',
    description: 'what a nice teapot display',
    url: '/../public/img/teapot-display.jpg'
  }
];

describe('Thumbnail', () => {
  it('renders the Thumbnail component', () => {
    const component = renderer.create(<Thumbnail images={images}/>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});