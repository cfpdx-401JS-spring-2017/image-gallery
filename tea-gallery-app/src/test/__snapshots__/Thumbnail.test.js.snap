// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail renders the Thumbnail component 1`] = `
<div>
  <ul
    className="thumbnail-list"
  >
    <div
      className="thumbnail-wrapper"
    >
      <li
        className="thumbnail-li"
      >
        <img
          alt="tea"
          className="thumbnail"
          src="/../public/img/teainpot.jpg"
        />
      </li>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
    <div
      className="thumbnail-wrapper"
    >
      <li
        className="thumbnail-li"
      >
        <img
          alt="tea"
          className="thumbnail"
          src="/../public/img/tetsubin-charcoal.jpg"
        />
      </li>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
    <div
      className="thumbnail-wrapper"
    >
      <li
        className="thumbnail-li"
      >
        <img
          alt="tea"
          className="thumbnail"
          src="/../public/img/teapot-display.jpg"
        />
      </li>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </ul>
</div>
`;
